apps/datagouvfr/lib/datagouvfr/client/datasets.ex

```diff
@@ -71,11 +65,62 @@ defmodule Datagouvfr.Client.Datasets do
     ]
   )
 
-  @spec new(map()) :: %__MODULE__{}
+  @callback new(map()) :: %__MODULE__{}
+  def new(map), do: impl().new(map)
+
+  @callback get_id_from_url(String.t()) :: String.t() | nil
+  def get_id_from_url(url), do: impl().get_id_from_url(url)
+
+  @callback get_infos_from_url(String.t()) :: map() | nil
+  def get_infos_from_url(url), do: impl().get_infos_from_url(url)
+
+  @doc """
+  Make a user follow a dataset
+  """
+  @callback post_followers(Plug.Conn.t(), String.t()) :: {atom, map}
+  def post_followers(%Plug.Conn{} = conn, dataset_id), do: impl().post_followers(conn, dataset_id)
+
+  @doc """
+  Make a user unfollow a dataset
+  """
+  @callback delete_followers(Plug.Conn.t(), String.t()) :: {atom, map}
+  def delete_followers(%Plug.Conn{} = conn, dataset_id), do: impl().delete_followers(conn, dataset_id)
+
+  @doc """
+  Fetch **only user IDs** following a dataset.
+  """
+  @callback get_followers(String.t()) :: {atom, map}
+  def get_followers(dataset_id), do: impl().get_followers(dataset_id)
+
+  @doc """
+  Call to GET /api/1/datasets/:id/
+  You can see documentation here: http://www.data.gouv.fr/fr/apidoc/#!/datasets/put_dataset
```

Comment:
```suggestion
  You can see documentation here: https://doc.data.gouv.fr/api/reference/#/datasets/put_dataset
```
Tu peux remplacer les autres liens aussi ?